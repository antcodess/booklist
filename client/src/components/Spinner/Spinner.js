import React from 'react';

import './Spinner.scss';

export const Spinner = () => {
  return (
    <div className='spinner-overlay'>
      <div className='spinner'></div>
    </div>
  );
};

export default Spinner;