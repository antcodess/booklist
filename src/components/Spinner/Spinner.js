import React from 'react';

import './Spinner.scss';

const Spinner = WrappedComponent => ({ isLoading, ...otherProps }) => {
  return isLoading ? (
    <div className='spinner-overlay'>
      <div className='spinner'></div>
    </div>
  ) : (
    <WrappedComponent {...otherProps} />
  )
};

export default Spinner;