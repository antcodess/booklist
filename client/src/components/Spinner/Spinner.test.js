import React from 'react';
import { shallow } from 'enzyme';
import { Spinner } from './Spinner';

describe('Spinner Component', () => {
  let wrapper;
  
  beforeEach(() => {
    wrapper = shallow(<Spinner />);
  });

  it('should render Spinner Component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});