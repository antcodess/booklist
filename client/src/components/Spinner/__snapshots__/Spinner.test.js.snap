// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner Component should render Spinner Component 1`] = `ShallowWrapper {}`;
